src/button-dropdown/interfaces.ts

```diff
@@ -136,6 +136,7 @@ export namespace ButtonDropdownProps {
     iconName?: IconProps.Name;
     iconUrl?: string;
     iconSvg?: React.ReactNode;
+    badge?: boolean;
```

Comment:
With this addition, it becomes a public API of the button-dropdown component too.

Fine with me, can be a nice feature, but it needs team consensus. Separate API proposal is not needed, but a brief discussion would be useful

Reply:
Just to clarify, after merging this PR, it becomes a defacto public API (not a beta like the drawers, but fully public).

This is why the discussion needs to happen before the merge

Reply:
It wasn't my intention to make it public. I feel like more conversations need to happen before we release this feature in a button dropdown, how can I make this a private feature?

Reply:
There is no easy way to make it public. In the past we used to have a private `href` feature (needed internally for breadcrumb group, but we did not make it public). Eventually, it leaked anyway and we had to call it de-facto public API. See AWSUI-12000 more details

I think, making it public is easier than setting up a guards against leaking this. I will try to help you with the team discussions